gallery-so/gallery
# [GAL-4095] [GAL-4114] Enable Sign Up with Wallet

### Summary of Changes

- User able to sign up with web3 wallet
- Moved some authentication hooks from `web` to `shared` 

### Demo

**Loom Video**
https://www.loom.com/share/dfb1cfcfbfd3424b9cc7ccfb24375bb9?sid=a81f55d9-d305-4020-a343-ad139094f3ed

### Edge Cases

1. Sign up with same username
2. Sign in with an existing account (should be redirect to feed and without welcome bottom sheet)
3. After enter the username, skip the pfp and bio

### Testing Steps

1. Connect you wallet (if the wallet has an existing account, you can delete it in dev db for testing purpose or use fresh wallet)
2. Enter your username
3. (optional) pick any pfp and enter your bio
4. should redirect to feed with welcome bottom sheet

### Checklist

Please make sure to review and check all of the following:
